core/validation/failures/user.py

```diff
@@ -295,10 +295,11 @@ def failures_to_messages(
 ) -> dict[str, list[str]] | dict[str, list[dict[str, str | None]]]:
     """Serialises failures into messages, removing any duplicates, and groups them by tab and section.
     :param validation_failures: validation failure objects
-    :return: validation failure messages grouped by tab and section
+    :return: validation failure messages grouped by tab and section and list of error types
     """
     # filter and convert to error messages
     error_messages = [failure.to_message() for failure in validation_failures]
+    error_types = [failure.__class__.__name__ for failure in validation_failures]
```

Comment:
I think some test coverage on this would be good - even if it's only a relatively small piece of code